core/src/test/scala/unit/kafka/log/LogCleanerIntegrationTest.scala

```diff
@@ -86,11 +86,13 @@ class LogCleanerIntegrationTest(compressionCodec: String) {
     // simulate deleting a partition, by removing it from logs
     // force a checkpoint
     // and make sure its gone from checkpoint file
-    cleaner.logs.remove(topics(0))
+    val removedLog = cleaner.logs.remove(topicPartitions(0))
+    removedLog.close()
```

Comment:
Maybe it's more robust to register the logs when we create them via `makeCleaner` and close them all during `tearDown`? Same for the other test.